libcontainer/standard_init_linux.go

```diff
@@ -260,6 +260,7 @@ func (l *linuxStandardInit) Init() error {
 		return &os.PathError{Op: "write exec fifo", Path: fifoPath, Err: err}
 	}
 
+	_ = unix.Close(fd)
```

Comment:
I don't think it's a bug. If you look at the code, you'll see that the parent process uses `handleFifoResult` to read all data from the fifo and (once it's closed from the other side) remove it.

Now, the presence of `exec.fifo` file is used to distinguish between the created and the running state, as you can see in `(*container).RefreshState` function.

Reply:
Obviously, we don't have any tests to check if the container state is reported correctly for the case when the startContainer hook is still running (which is what this PR essentially breaks).

If you have some time, feel free to write such a test @ningmingxiao

Reply:
Is it necessary to call "unix.Close(fd)" before to run startcontainer hook?

Reply:
> Is it necessary to call "unix.Close(fd)" before to run startcontainer hook?

No, it's not necessary. As I explained above, it only affects that the container status will be. Once it's closed, the `runc status` will show "running". This is why, I think, this commit is a mistake.

Reply:
thank you.